road/stats.py

```diff
@@ -0,0 +1,306 @@
+        relative_error_df = reorder_dataframe(relative_error_df, group_var)
+        est_obs_ratio_df = reorder_dataframe(est_obs_ratio_df, group_var)
+
+        # Count for each group_var category
+        if group_var == 'Observed Volume':
+            df['Observed Volume Category'] = df['Observed Volume'].apply(
+                classify_observation_volume)
+            counts = df['Observed Volume Category'].value_counts()
+            count_df = pd.DataFrame(counts).reset_index()
+            count_df.columns = ['Observed Volume', 'Count']
+        else:
+            counts = df[group_var].value_counts()
+            count_df = pd.DataFrame(counts).reset_index()
+            count_df.columns = [group_var, 'Count']
+
+        # Handle NaN values
+        if group_var in ['AT Group', 'FT Group']:
+            count_df[group_var] = count_df[group_var].fillna('Other')
+
+        count_df = count_df.groupby(group_var, as_index=False)['Count'].sum()
+
+        total_count = count_df['Count'].sum()
+        total_row = pd.DataFrame(
+            {group_var: ['All Locations'], 'Count': [total_count]})
+        count_df = pd.concat([count_df, total_row], ignore_index=True)
+        count_df = reorder_dataframe(count_df, group_var)
+
+        # Round the dataframes for better readability
+        percent_rmse_df = percent_rmse_df.round(1)
+        relative_error_df = relative_error_df.round(5)
+        est_obs_ratio_df = est_obs_ratio_df.round(3)
+
+        # Save the dataframes to CSV
+        group_var_no_spaces = group_var.replace(" ", "").lower()
+        file_prefix = f"{group_var_no_spaces}_"
+        count_df.to_csv(f'{file_prefix}count.csv', index=False)
+        percent_rmse_df.to_csv(f'percent_rmse_{group_var}.csv')
+        relative_error_df.to_csv(f'relative_error_{group_var}.csv')
+        est_obs_ratio_df.to_csv(f'est_obs_ratio_{group_var}.csv')
+
+        # Melt dataframes for easier plotting or analysis
+        melted_percent_rmse_df = percent_rmse_df.melt(
+            id_vars=[group_var], var_name='Time Period', value_name='Percent RMSE')
+        melted_relative_error_df = relative_error_df.melt(
+            id_vars=[group_var], var_name='Time Period', value_name='Relative Error')
+        melted_est_obs_ratio_df = est_obs_ratio_df.melt(
+            id_vars=[group_var], var_name='Time Period', value_name='Est/Obs Ratio')
+
+        # Save melted dataframes to CSV
+        melted_percent_rmse_df.to_csv(
+            f'{file_prefix}percent_rmse_melted.csv', index=False)
+        melted_relative_error_df.to_csv(
+            f'{file_prefix}relative_error_melted.csv', index=False)
+        melted_est_obs_ratio_df.to_csv(
+            f'{file_prefix}est_obs_ratio_melted.csv', index=False)
+
+        # generate the vega-lite files
+        generate_and_save_vega_lite_configs(group_var, file_prefix)
+
+
+def generate_and_save_vega_lite_configs(group_var, file_prefix):
+    output_dir = os.getcwd()

```

Comment:
this can be bad, e.g. if you're calling the scripts from another directory: e.g. `python valroad.py` and `python road/valroad.py` would give different results

Definitely pull the filepath or dir out as an input to this function and cascade it up to the wrapper validation_road.py script